src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -151,15 +151,26 @@ export const useSwapper = () => {
     if (!swapper) throw new Error('no swapper available')
     if (!wallet) throw new Error('no wallet available')
 
-    const result = await swapper.buildTrade({
-      sellAmount: amount,
-      sellAsset,
-      buyAsset,
-      sellAssetAccountNumber: 0, // TODO: remove hard coded accountId when multiple accounts are implemented
-      buyAssetAccountNumber: 0, // TODO: remove hard coded accountId when multiple accounts are implemented
-      wallet,
-      sendMax: true,
-    })
+    const result = await (async () => {
+      if (sellAsset.chainId === 'eip155:1') {
```

Comment:
we also have these constants for these chain ids https://github.com/shapeshift/lib/blob/main/packages/caip/src/constants.ts#L10

Reply:
Type 'string' is not assignable to type '"eip155:1" | "bip122:000000000019d6689c085ae165831e93"'.

Reply:
I wonder if we used the constants in the lib swapper types it would be able to narrow correctly. I guess this is one issue with string union discriminators vs string type variables..